sqlite3.go

```diff
@@ -719,93 +799,142 @@ func (c *SQLiteConn) RegisterAggregator(name string, impl any, pure bool) error
 	}
 
 	agg := t.Out(0)
+	var implReturnsPointer bool
 	switch agg.Kind() {
-	case reflect.Ptr, reflect.Interface:
+	case reflect.Ptr:
+		implReturnsPointer = true
+	case reflect.Interface:
+		implReturnsPointer = false
```

Comment:
1. How can you conclude whether it returns a pointer or not in this case?
2. Saying it doesn't return a pointer and allowing that seems to contradict the error message in the default case.

Reply:
You're right that I can't conclude if it returns a pointer or not in this case.

It does contradict the error message in the default case, but it was previously allowed in the initial implementation of user-defined functions #229.

Perhaps there is a better name for this variable.